domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1158,67 +1176,47 @@ private void doDelete(SQLConnection connection, boolean transactionMode, String
       String fieldName, String where, boolean returnCount, boolean returnIdField, boolean setId,
       List<FacetField> facets, Handler<AsyncResult<Results<T>>> replyHandler) {
 
-    long start = System.nanoTime();
-
     vertx.runOnContext(v -> {
       try {
         String addIdField = "";
         if (returnIdField) {
-          addIdField = "," + idField;
+          addIdField = COMMA + idField;
         }
 
         if (!"null".equals(fieldName) && fieldName.contains("*")) {
-          //if we are requesting all fields (*) , then dont add the id field to the select
-          //this will return two id columns which will create ambiguity in facet queries
+          // if we are requesting all fields (*) , then dont add the id field to the select
+          // this will return two id columns which will create ambiguity in facet queries
           addIdField = "";
         }
 
-        String from2where = FROM + convertToPsqlStandard(tenantId) + "." + table + " " + where;
+        String[] q = new String[2];
```

Comment:
Consider replacing q by two variables with better names, for example selectQuery and countQuery. Change function interface of processQueryWithCount accordingly.

Reply:
This refactoring makes sense. I will get these changes in.